tests/test_data.py

```diff
@@ -46,3 +46,39 @@ def test_looks_like_test_file_false(filename):
         from_imports={},
     )
     assert not state.looks_like_test_file()
+
+
+@pytest.mark.parametrize(
+    ("filename"),
+    (
+        "__init__.py",
+        "package/__init__.py",
+        "project/package/__init__.py",
+    ),
+)
+def looks_like_dunder_init_file_true(filename):
+    state = State(
+        settings=Settings(target_version=(4, 0)),
+        filename=filename,
+        from_imports={},
+    )
+    assert state.looks_like_test_file()
+
+
+@pytest.mark.parametrize(
+    ("filename"),
+    (
+        "__thing__init__.py",
```

Comment:
for the word separator change

```suggestion
        "__thing__init__.py",
        "thing-__init__.py",
```